docs/fundamentals/as-avs.txt

```diff
@@ -0,0 +1,262 @@
+- :atlas:`Atlas Search Overview </atlas-search/atlas-search-overview/>`
+- :atlas:`Atlas Vector Search Overview </atlas-vector-search/vector-search-overview/>`
+
+.. note:: Deployment Compatibility
+
+   You can use the Atlas Search and Vector Search features only when
+   connecting to MongoDB Atlas clusters. These features are not available for
+   self-managed deployments.
+
+This page describes how to create the appropriate indexes for each type
+of search and provides examples of how to use the {+odm-short+} to
+perform searches.
+
+Atlas Search
+------------
+
+In this section, you can learn how to create Atlas Search indexes
+and perform searches in the {+odm-short+}. The Atlas Search API
+internally uses the ``$search`` aggregation operator to perform queries.
+To learn more about this operator, see the :atlas:`$search
+</atlas-search/aggregation-stages/search/>` reference in the Atlas documentation.
+
+.. _laravel-as-index:
+
+Create an Atlas Search Index
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. TODO in DOCSP-46230
+
+Perform Queries
+~~~~~~~~~~~~~~~
+
+In this section, you can learn how to use the Atlas Search API in the
+{+odm-short+}.
+
+General Queries
+```````````````
+
+The {+odm-short+} provides the ``search()`` method as a query
+builder method and as an Eloquent model method. You can use the
+``search()`` method to run Atlas Search queries on documents in your
+collections.
+
+You must pass an ``operator`` parameter to the ``search()`` method that
+is an instance of ``SearchOperatorInterface`` or an array. You can
+create an instance of ``SearchOperatorInterface`` by calling the
+``Search::text()`` method and passing the field you are
+querying and your search term or phrase.
+
+You must include the following import statement in your application to
+create a ``SearchOperatorInterface`` instance:
+
+.. code-block:: php
+
+   use MongoDB\Builder\Search;
+
+The following code uses the index created in the preceding
+:ref:`laravel-as-index` section to perform an Atlas Search query on the
+``movies`` collection:
+
+.. io-code-block::
+   :copyable: true
+
+   .. input:: /includes/fundamentals/as-avs/AtlasSearchTest.php
+      :language: php
+      :dedent:
+      :start-after: start-search-query
+      :end-before: end-search-query
+
+   .. output::
+      :language: json
+      :visible: false
+
+      [
+         { "title": "Dreaming of Jakarta",
+           "year": 1990
+         },
+         { "title": "See You in My Dreams",
+           "year": 1996
+         }
+      ]
+
+You can use the ``search()`` method to perform many types of Atlas
+Search queries. Depending on your desired query, you can pass the
+following optional parameters to ``search()``:
+
+- ``index``: ``string``
+- ``highlight``: ``array``
+- ``concurrent``: ``bool``
+- ``count``: ``string``
+- ``searchAfter``: ``string``
+- ``searchBefore`` ``string``
+- ``scoreDetails``: ``bool``
+- ``sort``: ``array``
+- ``returnStoredSource``: ``bool``
+- ``tracking``: ``array``
```

Comment:
S: Would be good to have a table with a description of what these parameters do. Otherwise, I don't see the point of listing them when readers won't intuitively know what they do.

Reply:
I outsource these option definitions to the atlas documentation in the link after the list. THis way, we can avoid repeating the same info in two places

Reply:
Hmm, perhaps listing the names of the parameters without the data type would suffice then, since the atlas documentation also includes the data type. The data type seems like unnecessary info if I don't know what the parameter is for. I'll leave up to you though!